pygdbmi/gdbcontroller.py

```diff
@@ -85,6 +84,14 @@ def __init__(self, gdb_path='gdb', gdb_args=['--nx', '--quiet', '--interpreter=m
         self.read_list = [self.stdout_fileno, self.stderr_fileno]
         self.write_list = [self.stdin_fileno]
 
+        self.incomplete_packet = {} # string buffers for unifinished gdb output
```

Comment:
I like that you added another buffer for stderr. 

The word `packet` already has a meaning in gdb (https://sourceware.org/gdb/current/onlinedocs/gdb/Packets.html) so I think it should be avoided.

I would recommend creating and assigning the dictionary in one call:
```python
self._incomplete_output = {'stdout': None, 'stderr': None}
```